### PR TITLE
lambda function improvement, method signature and general warnings

### PR DESCRIPTION
**Pull request by zishanbilal**
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/798)

### CODE REVIEW

File: src/main/scala/beam/agentsim/agents/vehicles/BeamVehicleType.scala [hunk 1]
```diff
@@ -108,7 +108,7 @@ object BeamVehicleType {
     4,
     0,
     4.5,
-    new FuelType("gasoline", 0.0),
+    FuelType("gasoline", 0.0),
```
Comment (carloscaldas) [inline src/main/scala/beam/agentsim/agents/vehicles/BeamVehicleType.scala]: Consider using the suffix:
- Float: 0F
- Double: 0D

File: src/main/scala/beam/router/osm/TollCalculator.scala [hunk 2]
```diff
@@ -73,7 +70,7 @@ class TollCalculator(val directory: String) extends LazyLogging {
     }
 
     if (Files.isDirectory(directory)) {
-      directory.toFile.listFiles(hasOSM(_)).map(_.getAbsolutePath).headOption.foreach(loadOSM)
+      directory.toFile.listFiles(hasOSM).map(_.getAbsolutePath).headOption.foreach(loadOSM)
```
Comment (carloscaldas) [inline src/main/scala/beam/router/osm/TollCalculator.scala]: Consider using fluent approach in multiple lines

File: src/main/scala/beam/utils/SpatialPlot.scala [hunk 3]
```diff
@@ -253,7 +253,7 @@ object SpatialPlot extends App {
 
   val spatialPlot = new SpatialPlot(1000, 1000, 20)
 
-  for (i <- 1 until 100) {
+  for (_ <- 1 until 100) {
```
Comment (carloscaldas) [inline src/main/scala/beam/utils/SpatialPlot.scala]: replace the `for` with the range `1 until 100`

File: src/main/scala/beam/utils/matsim_conversion/MatsimPlanConversion.scala [hunk 4]
```diff
@@ -147,7 +147,7 @@ object MatsimPlanConversion {
         n match {
           case elem: Elem if elem.label == "person" =>
             val attrs = elem.attributes
-            val filteredAttrs = attrs.filter(_.key.equals("id"))
+            val filteredAttrs = attrs.filter(m => m.key.equals("id"))
```
Comment (JustinPihony) [inline src/main/scala/beam/utils/matsim_conversion/MatsimPlanConversion.scala]: As this is not using a meaningful name I see this as just adding white noise. Unless we are stepping away from using underscores altogether?

File: src/test/scala/beam/integration/ridehail/RideHailBufferedRidesSpec.scala [hunk 5]
```diff
@@ -25,9 +25,9 @@ class RideHailBufferedRidesSpec extends FlatSpec with BeamHelper with MockitoSug
     }
 
     groupedByPerson.map {
-      case (id, events) =>
+      case (id, es) =>
```
Comment (JustinPihony) [inline src/test/scala/beam/integration/ridehail/RideHailBufferedRidesSpec.scala]: `events` was more meaningful. I'd say to keep that verbiage

**Comment by zishanbilal** [comment]
@carloscaldas  please review the changes.

**Pull request merged**